### PR TITLE
Priority 1 tests for OSX & Linux x64 Release builds should now run daily

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

**Comment by wtgodbe** [comment]
@adityamandaleeka can you take a look?

**Comment by wtgodbe** [comment]
@dotnet-bot test this please

File: build.cmd [hunk 1]
```diff
@@ -49,6 +49,8 @@ if /i "%1" == "vs2015" (set __VSVersion=%1&shift&goto Arg_Loop)
 if /i "%1" == "skiptestbuild" (set __SkipTestBuild=1&shift&goto Arg_Loop)
 if /i "%1" == "docrossgen" (set __DoCrossgen=1&shift&goto Arg_Loop)
 
+if /i "%1" == "priority" (set _priorityvalue=%2&shift&shift&goto Arg_Loop)
```
Comment (adityamandaleeka) [inline build.cmd]: I think this should be called testpriority or something. Also, I'd start the variable name with two underscores and camel case it to match the convention here.

File: build.cmd [hunk 2]
```diff
@@ -272,7 +274,15 @@ if defined __SkipTestBuild (
 echo.
 echo Commencing build of tests for %__BuildOS%.%__BuildArch%.%__BuildType%
 echo.
+
+if defined _priorityvalue (
+    call "%__ProjectDir%\tests\buildtest.cmd" Priority %_priorityvalue%
+    goto ErrCheck
```
Comment (adityamandaleeka) [inline build.cmd]: I would make a variable to hold buildtest args, and if _priorityvalue is defined, add the priority stuff to it. That will allow you to remove this goto and have just one place where you call buildtest.

File: build.cmd [hunk 3]
```diff
@@ -301,4 +311,5 @@ echo linuxmscorlib - Build mscorlib for Linux
 echo osxmscorlib - Build mscorlib for OS X
 echo skiptestbuild - Skip building tests
 echo toolset_dir - toolset directory -- Arm64 use only.
+echo Priority (N) where N is a number greater than zero that signifies the set of tests that will be built and consequently run.
```
Comment (adityamandaleeka) [inline build.cmd]: Can it be 0? I assume that would just be equivalent to not specifying a priority at all, but it is a valid option right?

File: netci.groovy [hunk 4]
```diff
@@ -126,263 +126,298 @@ def static addPRTrigger(def job, def architecture, def os, def configuration, is
...
-        ['Ubuntu', 'OSX'].each { os ->
-            ['Debug', 'Release'].each { configuration ->
-                def lowerConfiguration = configuration.toLowerCase()
-                def osGroup = getOSGroup(os)
-                def jobName = getBuildJobName(configuration, architecture, os) + "_tst"
-                def inputCoreCLRBuildName = Utilities.getFolderName(project) + '/' + 
-                    Utilities.getFullJobName(project, getBuildJobName(configuration, architecture, os), isPR)
-                def inputWindowTestsBuildName = Utilities.getFolderName(project) + '/' + 
-                    Utilities.getFullJobName(project, getBuildJobName(configuration, architecture, 'windows_nt'), isPR)
+[true, false].each { buildPri1 ->
```
Comment (adityamandaleeka) [inline netci.groovy]: Here too

File: netci.groovy [hunk 5]
```diff
@@ -126,263 +126,298 @@ def static addPRTrigger(def job, def architecture, def os, def configuration, is
...
+    [true, false].each { isPR ->
+        // Architectures.  x64 only at this point
+        ['x64'].each { architecture ->
+            // Put the OS's supported for coreclr cross testing here
+            ['Ubuntu', 'OSX'].each { os ->
+                ['Debug', 'Release'].each { configuration ->
+                    // Skip unnecessary Pri1 legs
+                    if (buildPri1 && (!isPR || configuration != 'Release'))
```
Comment (adityamandaleeka) [inline netci.groovy]: Why is it !isPR?

File: netci.groovy [hunk 6]
```diff
@@ -126,263 +126,298 @@ def static addPRTrigger(def job, def architecture, def os, def configuration, is
...
-                    default:
-                        println("Unknown architecture: ${architecture}");
-                        assert false
-                        break
-                }
-                
-                // Calculate names
-                def lowerConfiguration = configuration.toLowerCase()
-                def jobName = getBuildJobName(configuration, architecture, os)
+[true, false].each { buildPri1 ->
```
Comment (adityamandaleeka) [inline netci.groovy]: 'buildPri1Tests'? The current name isn't really clear.

**Comment by adityamandaleeka** [comment]
Is there some documented understanding on what the various test priorities mean? If someone adds a bunch of tests with priorities between 0 and 5, some of them will be run in CI and others will not. Is the intention here to periodically run everything that's doesn't run in PRs/pushes because it's not pri 0, or is it specifically to run pri 1 stuff? Those sets might be the same today, but that could change.

**Comment by adityamandaleeka** [comment]
Also, please squash your commits prior to merge.

**Comment by wtgodbe** [comment]
I’m not sure what each priority level means, nor what the highest level is. That’s probably a specification that should exist if it doesn’t already – who might know more about this? An ugly stopgap might be to just run all tests up to some reasonably large priority number

**Pull request closed without merging**